Orange/widgets/visualize/tests/test_owsieve.py

```diff
@@ -22,10 +24,52 @@ def setUpClass(cls):
 
         cls.signal_name = "Data"
         cls.signal_data = cls.data
+        cls.titanic = Table("titanic")
+        cls.iris = Table("iris")
 
     def setUp(self):
         self.widget = self.create_widget(OWSieveDiagram)
 
+    def test_context_settings(self):
+        # Set titanic and check first two attributes on display
+        self.send_signal(self.widget.Inputs.data, self.titanic)
+        self.assertEqual(self.widget.attr_x, self.titanic.domain.class_var)
+        self.assertEqual(self.widget.attr_y, self.titanic.domain.attributes[0])
+        # Change attributes to two different ones
+        self.widget.attr_x = self.titanic.domain.attributes[1]
+        self.widget.attr_y = self.titanic.domain.attributes[2]
+        # Remove data signal
+        self.send_signal(self.widget.Inputs.data, None)
+        self.assertEqual(self.widget.attr_x, None)
+        self.assertEqual(self.widget.attr_y, None)
+        # Set data back to titanic and check if selected attributes are
+        # remembered in the settings
+        self.send_signal(self.widget.Inputs.data, self.titanic)
+        self.assertEqual(self.widget.attr_x, self.titanic.domain.attributes[1])
+        self.assertEqual(self.widget.attr_y, self.titanic.domain.attributes[2])
+
+    def test_continuous_data(self):
+        self.send_signal(self.widget.Inputs.data, self.iris)
+        self.assertEqual(self.widget.attr_x, self.iris.domain.class_var)
+        self.assertEqual(self.widget.attr_y, self.iris.domain.attributes[0])
+        self.assertTrue(self.widget.discrete_data)
```

Comment:
I'd prefer using `assertIsNotNone` here, since that's what you're actually testing. Yet, this test doesn't do much since `discrete_data` will be non-None whenever `data` is not `None`. Maybe you want to test

        self.assertTrue(self.widget.discrete_data.domain[0].is_discrete)